test/e2e/page-objects/pages/home/activity-list.ts

```diff
@@ -161,20 +161,36 @@ class ActivityListPage {
     await this.driver.assertElementNotPresent(this.completedTransactions);
   }
 
-  async checkTxAction(expectedAction: string, expectedNumber: number = 1) {
+  /**
+   * Check if a transaction at the specified index displays the expected action text in the activity list.
+   *
+   * @param params - The parameters object containing:
+   * @param params.action - The expected action text to be displayed (e.g., "Send", "Receive", "Swap")
+   * @param params.txIndex - The index of the transaction to check in the activity list
+   * @param params.completedTxs - The total number of completed transactions expected to be displayed in the activity list
+   * @returns A promise that resolves if the transaction at the specified index displays the expected action text within the timeout period.
+   */
+  async checkTxAction({
+    action,
+    txIndex = 1,
+    completedTxs = 1,
+  }: {
+    action: string;
+    txIndex?: number;
+    completedTxs?: number;
+  }): Promise<void> {
+    // We need to wait for the total number of tx's to be able to use getText() without race conditions.
+    await this.checkCompletedTxNumberDisplayedInActivity(completedTxs);
+
```

Comment:
:information_source: I've decided to go with this approach: I found that there's no way to make the getText() not flaky, unless we make sure that all the expected tx are in the activity list first. Adding this pre-step ensures that the function is always robust.
I found that if we assume this is done in a separate function (in the test) it can be cases where thischeck is not applied (bc the author doesn't know) and it's hard to make everyone aware. So I decided to go for this route, to just add the check here and then, the function can be used out of the box.
Let me know if this makes sense @chloeYue 

Thankyou for the great catch!!

Side: I added the params as object as it's easier to navigate, and JsDocs